packages/base-styles/_animations.scss

```diff
@@ -1,5 +1,37 @@
-@mixin edit-post__fade-in-animation($speed: 0.08s, $delay: 0s) {
-	animation: edit-post__fade-in-animation $speed linear $delay;
+@mixin keyframes($name) {
+	@keyframes #{$name} {
+		@content;
+	}
+}
+
+@mixin animation__fade-in($speed: 0.08s, $delay: 0s) {
+	@include keyframes(__wp-base-styles-fade-in) {
+		from {
+			opacity: 0;
+		}
+		to {
+			opacity: 1;
+		}
+	}
+
+
+	animation: __wp-base-styles-fade-in $speed linear $delay;
+	animation-fill-mode: forwards;
+	@include reduce-motion("animation");
```

Comment:
What are your thoughts on the continued usage of this mixin? For new code, should we just write out the media query like we're doing in the components package?

Reply:
I like the approach that we're taking in components package, ie. "only add animation styles if the user does NOT prefer reduced motion".

Having said that, I would prefer not to make such a change here for two reasons:
- consistency: if make such a change, we could consider doing so at the project level?
- the `reduce-motion` mixin sets durations to `0s`, which is technically different from not having an animation at all. For example, I'm not sure how animation and transition events would fire across browsers, and that could introduce unexpected behaviors.

To sum it up: I'd be up for making the change, but I'd make it separately from this PR and gradually:
- refactor usages one by one to `@media not (prefers-reduce-motion) { /* animation code /* }`, test for breaking changes
- when all usages are migrated, deprecate the mixing

Reply:
agree with @ciampo, worth doing but with precaution

Reply:
Got it 👍

Reply:
That said, regarding this:

> the reduce-motion mixin sets durations to 0s, which is technically different from not having an animation at all. For example, I'm not sure how animation and transition events would fire across browsers, and that could introduce unexpected behaviors

I'm not 100% sure, but I can't see how a duration of 0 would be effectively any different from disabling the animation itself, other than if there's a delay in one of them or something like that which I doubt. Btw, now that I think about it that approach (setting duration to 0) was flawed in that way. A delay would still affect it. One more reason to migrate away from that pattern.

Reply:
FWIW the duration is set to 1ms, which may be even more different 😅